src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/OpenDocumentGenerator.cs

```diff
@@ -0,0 +1,317 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal class OpenDocumentGenerator : ProjectSnapshotChangeTrigger
+    {
+        private readonly ForegroundDispatcher _foregroundDispatcher;
+        private readonly IEnumerable<DocumentProcessedListener> _documentProcessedListeners;
+        private readonly Dictionary<string, DocumentSnapshot> _work;
+        private ProjectSnapshotManagerBase _projectManager;
+        private Timer _timer;
```

Comment:
It looks like we never do anything with the timer but dispose it?

Reply:
[`StartWorker`](https://github.com/dotnet/aspnetcore-tooling/pull/3309/files#diff-f7c53260e5d017b0f3f40790a21a2a353878592ba91bc85b6af81458b44c6453R162) is the only piece that actually news it up. It's effectively a "fire once in another task/thread" model. Timers at their core are better used for this type of quick work; however, it's definitely a little odd. lol it's effectively like `Task.Run(() => ...)` except we use the timers dedicated pool to run. Honestly if there wasn't a precedent for our use of timers I'd probably not use them at all; however, this class is mostly a copy from the pre-existing BackgroundDOcumentGenerator.